packages/playwright-core/src/server/recorder.ts

```diff
@@ -178,7 +178,8 @@ export class Recorder implements InstrumentationListener {
     });
 
     await this._context.exposeBinding('__pw_recorderSetSelector', false, async (_, selector: string) => {
-      await this._recorderApp?.setSelector(selector, true);
+      const selectorWithFrame = _.frame.parentFrame() ? `iframe >> nth=0 >> internal:control=enter-frame >> ${selector}` : selector;
```

Comment:
> iframe >> nth=0

That's not accurate - we would need exact frame locator along with the tests for this feature!

Reply:
Yes, since I can not find any locator for iframe, I just hard code `iframe >> nth=0`.
Will we pass the frame locator within `selector` or a new argument?

Reply:
I'm not sure I understand the question!